packages/tgc/src/common.components/search-input/search-input.tsx

```diff
@@ -0,0 +1,137 @@
+import { useLazyGetDisplayNameSuggestionsQuery } from 'platform-apis';
+import { IconSearch, Input, LINKS, NOTIFICATIONS_DURATION, SNACKBAR_TYPE } from 'platform-components';
+import React, { useEffect, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import { useLocation } from 'wouter';
+
+import { clearChannelsState } from '../../store/slices/channels';
+import { clearMessages } from '../../store/slices/messages';
+import { getUserTypeSetting } from '../../store/slices/settings';
+import {
+    clearSuggestions, clearTwitchUser,
+    getIsSuggestionsLoading,
+    getIsTwitchUserFetching,
+    getSuggestions,
+} from '../../store/slices/twitch-user';
+import { getIsAuth } from '../../store/slices/user';
+import { addNotification, isValidSearchChange, isValidSearchSubmit } from '../../utils';
+
+import SearchInputSuggestions from './__suggestions/search-input__suggestions';
+
+const SearchInput = () => {
+    const dispatch = useDispatch();
+    const [usernameParam] = useState('');
+    const isAuth = useSelector(getIsAuth);
+
+    const intl = useIntl();
+    const [location, navigate] = useLocation();
+
+    const suggestions = useSelector(getSuggestions);
+    const isSuggestionsLoading = useSelector(getIsSuggestionsLoading);
+
+    const isSearchingUser = useSelector(getIsTwitchUserFetching);
+
+    const [typingTimer, setTypingTimer] = useState<NodeJS.Timeout | null>(null);
+    const [username, setUsername] = useState(usernameParam || '');
```

Comment:
therefore here we need to rename `username, setUsername` to `displayName` to avoid confusion and as the initial parameter value from props is the best choice